CMakeLists.txt

```diff
@@ -54,17 +54,36 @@
 
 cmake_minimum_required(VERSION 3.1.0)
 
-set(CMAKE_MODULE_PATH Builds/CMake)
+if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
+  set(dir "build")
+  set(cmd "cmake")
+  if (target)
+    set(dir "${dir}/${target}")
+    set(cmd "${cmd} -Dtarget=${target}")
+  elseif(CMAKE_BUILD_TYPE)
+    set(dir "${dir}/${CMAKE_BUILD_TYPE}")
+    set(cmd "${cmd} -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}")
+  else()
+    set(dir "${dir}/default")
+  endif()
+  set(cmd "${cmd} ../..")
+
+  message(FATAL_ERROR "In-source builds are not allowed. Instead try:\n"
+    "rm CMakeCache.txt && mkdir -p ${dir} && cd ${dir} && ${cmd}")
```

Comment:
I discovered via a little experimentation that it was necessary and sufficient to delete `CMakeCache.txt` from the in-source folder to allow out-of-source builds. I _could_ suggest `rm -r CMakeFiles CMakeCache.txt`, but I figured caution was a little more prudent. Given that, what do you think?